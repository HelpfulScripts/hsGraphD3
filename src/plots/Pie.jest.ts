import * as hsGraphD3   from '../';
import * as d3          from 'd3';
import { DataSet }      from '../Graph';

let clientWidth = 300;
const root = window.document.createElement("div");
root.style.width = `${clientWidth}px`;
root.style.height = "300px";

d3.selection.prototype.transition   = function(){ return this; };
d3.selection.prototype.duration     = function(){ return this; };
d3.selection.prototype.ease         = function(){ return this; };

const data:DataSet = {
    colNames:['time', 'volume', 'costs'], 
    rows:[    [-1,    0.2,      0.3], 
              [0.2,   0.7,      0.2], 
              [0.4,   0.1,      0.3],
              [0.6,  -0.2,      0.1], 
              [0.8,   0.3,      0.5], 
              [1,     0.2,      0.4]
    ]
};

function createGraph(root:any) {
    const graph = new hsGraphD3.Graph(root);
    graph.isRendered = () => true;
    return graph;
}


describe('Pie labels', () => {
    let graph:hsGraphD3.Graph;
    beforeAll(() => {
        graph = createGraph(root);
        graph.add('pie', {phi:'costs', label:(row, i)=>i});
        graph.add('pie', {phi:'costs', label:(row, i)=>i});
        graph.defaults.series[0].label.xpos = 0.5;
        graph.title.text = `Pie Graph`;
        graph.render(data);
    });
    it('should plot pie', () => expect(root).toMatchSnapshot());
});

describe('Pie colors', () => {
    let graph:hsGraphD3.Graph;
    beforeAll(() => {
        graph = createGraph(root);
        graph.add('pie', {phi:'costs', r0: 5, color:'time'});
        graph.render(data);
    });
    it('should plot pie', () => expect(root).toMatchSnapshot());
});
