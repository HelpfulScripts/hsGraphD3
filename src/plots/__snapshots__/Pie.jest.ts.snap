// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pie colors should plot pie 1`] = `
<div
  style="width: 300px; height: 300px;"
>
  <svg
    class="baseSVG"
    height="100%"
    preserveAspectRatio="xMinYMin meet"
    viewBox="-500 -350 1000 700"
    width="100%"
  >
    <g
      class="canvas"
    >
      <rect
        class="graphArea"
        fill="#fff"
        fill-opacity="1"
        height="700"
        rx="0"
        ry="0"
        stroke="#00c"
        stroke-dasharray="none"
        stroke-opacity="1"
        stroke-width="1"
        width="1000"
        x="-500"
        y="-350"
      />
    </g>
    <g
      class="grids"
    >
      <g
        class="hor-major-Grid"
      />
      <g
        class="hor-minor-Grid"
      />
      <g
        class="ver-major-Grid"
      />
      <g
        class="ver-minor-Grid"
      />
    </g>
    <g
      class="axes"
    >
      <g
        class="horAxis"
      />
      <g
        class="verAxis"
      />
    </g>
    <g
      class="title"
    >
      <text
        dominant-baseline="hanging"
        fill="currentColor"
        font-family="sans-serif"
        font-size="24px"
        font-style="normal"
        font-weight="bold"
        text-anchor="start"
        x="-450"
        y="-315"
      >
        Pie Graph
      </text>
    </g>
    <g
      class="series"
    >
      <g
        class="series0"
        style="color: rgb(27, 158, 119);"
      >
        <g
          class="markers"
          fill="currentColor"
          fill-opacity="0.75"
          stroke="currentColor"
          stroke-dasharray="none"
          stroke-opacity="1"
          stroke-width="1"
        >
          <path
            class="popup"
            d="M-1.2442848271512585e-13,325A325,325,0,0,1,-281.45825622994266,162.49999999999986L0,0Z"
          />
          <path
            class="popup"
            d="M-281.4582562299424,-162.50000000000028A325,325,0,0,1,-111.156546580842,-305.40010175542034L0,0Z"
          />
          <path
            class="popup"
            d="M-281.45825622994266,162.49999999999986A325,325,0,0,1,-281.4582562299424,-162.50000000000028L0,0Z"
          />
          <path
            class="popup"
            d="M-111.156546580842,-305.40010175542034A325,325,0,0,1,5.176144413466479e-13,-325L0,0Z"
          />
          <path
            class="popup"
            d="M1.990051048614449e-14,-325A325,325,0,0,1,320.06251972896763,56.435657741752436L0,0Z"
          />
          <path
            class="popup"
            d="M320.06251972896763,56.435657741752436A325,325,0,0,1,-1.2442848271512585e-13,325L0,0Z"
          />
        </g>
        <g
          class="label"
          fill="#000"
          font-family="sans-serif"
          font-size="16px"
          font-style="normal"
          font-weight="100"
        >
          <text
            class="popup"
            dominant-baseline="middle"
            dx="0.0em"
            dy="0.0em"
            text-anchor="middle"
            transform="translate(-81.25000000000004,140.72912811497127)"
          >
            0.0
          </text>
          <text
            class="popup"
            dominant-baseline="middle"
            dx="0.0em"
            dy="0.0em"
            text-anchor="middle"
            transform="translate(-104.45298657406255,-124.482222006834)"
          >
            1.0
          </text>
          <text
            class="popup"
            dominant-baseline="middle"
            dx="0.0em"
            dy="0.0em"
            text-anchor="middle"
            transform="translate(-162.5,-1.2442848271512585e-13)"
          >
            2.0
          </text>
          <text
            class="popup"
            dominant-baseline="middle"
            dx="0.0em"
            dy="0.0em"
            text-anchor="middle"
            transform="translate(-28.217828870875895,-160.03125986448387)"
          >
            3.0
          </text>
          <text
            class="popup"
            dominant-baseline="middle"
            dx="0.0em"
            dy="0.0em"
            text-anchor="middle"
            transform="translate(124.48222200683395,-104.45298657406263)"
          >
            4.0
          </text>
          <text
            class="popup"
            dominant-baseline="middle"
            dx="0.0em"
            dy="0.0em"
            text-anchor="middle"
            transform="translate(104.45298657406259,124.48222200683396)"
          >
            5.0
          </text>
        </g>
      </g>
      <g
        class="series1"
        style="color: rgb(217, 95, 2);"
      >
        <g
          class="markers"
          fill="currentColor"
          fill-opacity="0.75"
          stroke="currentColor"
          stroke-dasharray="none"
          stroke-opacity="1"
          stroke-width="1"
        >
          <path
            class="popup"
            d="M-1.2442848271512585e-13,325A325,325,0,0,1,-281.45825622994266,162.49999999999986L0,0Z"
          />
          <path
            class="popup"
            d="M-281.4582562299424,-162.50000000000028A325,325,0,0,1,-111.156546580842,-305.40010175542034L0,0Z"
          />
          <path
            class="popup"
            d="M-281.45825622994266,162.49999999999986A325,325,0,0,1,-281.4582562299424,-162.50000000000028L0,0Z"
          />
          <path
            class="popup"
            d="M-111.156546580842,-305.40010175542034A325,325,0,0,1,5.176144413466479e-13,-325L0,0Z"
          />
          <path
            class="popup"
            d="M1.990051048614449e-14,-325A325,325,0,0,1,320.06251972896763,56.435657741752436L0,0Z"
          />
          <path
            class="popup"
            d="M320.06251972896763,56.435657741752436A325,325,0,0,1,-1.2442848271512585e-13,325L0,0Z"
          />
        </g>
        <g
          class="label"
          fill="#000"
          font-family="sans-serif"
          font-size="16px"
          font-style="normal"
          font-weight="100"
        >
          <text
            class="popup"
            dominant-baseline="middle"
            dx="0.0em"
            dy="0.0em"
            text-anchor="middle"
            transform="translate(-81.25000000000004,140.72912811497127)"
          >
            0.0
          </text>
          <text
            class="popup"
            dominant-baseline="middle"
            dx="0.0em"
            dy="0.0em"
            text-anchor="middle"
            transform="translate(-104.45298657406255,-124.482222006834)"
          >
            1.0
          </text>
          <text
            class="popup"
            dominant-baseline="middle"
            dx="0.0em"
            dy="0.0em"
            text-anchor="middle"
            transform="translate(-162.5,-1.2442848271512585e-13)"
          >
            2.0
          </text>
          <text
            class="popup"
            dominant-baseline="middle"
            dx="0.0em"
            dy="0.0em"
            text-anchor="middle"
            transform="translate(-28.217828870875895,-160.03125986448387)"
          >
            3.0
          </text>
          <text
            class="popup"
            dominant-baseline="middle"
            dx="0.0em"
            dy="0.0em"
            text-anchor="middle"
            transform="translate(124.48222200683395,-104.45298657406263)"
          >
            4.0
          </text>
          <text
            class="popup"
            dominant-baseline="middle"
            dx="0.0em"
            dy="0.0em"
            text-anchor="middle"
            transform="translate(104.45298657406259,124.48222200683396)"
          >
            5.0
          </text>
        </g>
      </g>
    </g>
    <g
      class="popup"
    />
    <style>
      .popup:hover { filter: brightness(0.8);}
    </style>
  </svg>
  <svg
    class="baseSVG"
    height="100%"
    preserveAspectRatio="xMinYMin meet"
    viewBox="-500 -350 1000 700"
    width="100%"
  >
    <g
      class="canvas"
    >
      <rect
        class="graphArea"
        fill="#fff"
        fill-opacity="1"
        height="700"
        rx="0"
        ry="0"
        stroke="#00c"
        stroke-dasharray="none"
        stroke-opacity="1"
        stroke-width="1"
        width="1000"
        x="-500"
        y="-350"
      />
    </g>
    <g
      class="grids"
    >
      <g
        class="hor-major-Grid"
      />
      <g
        class="hor-minor-Grid"
      />
      <g
        class="ver-major-Grid"
      />
      <g
        class="ver-minor-Grid"
      />
    </g>
    <g
      class="axes"
    >
      <g
        class="horAxis"
      />
      <g
        class="verAxis"
      />
    </g>
    <g
      class="title"
    >
      <text />
    </g>
    <g
      class="series"
    >
      <g
        class="series0"
        style="color: rgb(27, 158, 119);"
      >
        <g
          class="markers"
          fill="currentColor"
          fill-opacity="0.75"
          stroke="currentColor"
          stroke-dasharray="none"
          stroke-opacity="1"
          stroke-width="1"
        >
          <path
            class="popup"
            d="M-1.2442848271512585e-13,325A325,325,0,0,1,-281.45825622994266,162.49999999999986L-56.29165124598853,32.49999999999997A65,65,0,0,0,-2.488569654302517e-14,65Z"
          />
          <path
            class="popup"
            color="#2ca02c"
            d="M-281.4582562299424,-162.50000000000028A325,325,0,0,1,-111.156546580842,-305.40010175542034L-22.231309316168403,-61.080020351084066A65,65,0,0,0,-56.29165124598848,-32.50000000000006Z"
          />
          <path
            class="popup"
            color="#9467bd"
            d="M-281.45825622994266,162.49999999999986A325,325,0,0,1,-281.4582562299424,-162.50000000000028L-56.29165124598848,-32.50000000000006A65,65,0,0,0,-56.29165124598853,32.49999999999997Z"
          />
          <path
            class="popup"
            color="#e377c2"
            d="M-111.156546580842,-305.40010175542034A325,325,0,0,1,5.176144413466479e-13,-325L1.0352288826932959e-13,-65A65,65,0,0,0,-22.231309316168403,-61.080020351084066Z"
          />
          <path
            class="popup"
            color="#bcbd22"
            d="M1.990051048614449e-14,-325A325,325,0,0,1,320.06251972896763,56.435657741752436L64.01250394579353,11.287131548350487A65,65,0,0,0,3.980102097228898e-15,-65Z"
          />
          <path
            class="popup"
            d="M320.06251972896763,56.435657741752436A325,325,0,0,1,-1.2442848271512585e-13,325L-2.488569654302517e-14,65A65,65,0,0,0,64.01250394579353,11.287131548350487Z"
          />
        </g>
      </g>
    </g>
    <g
      class="popup"
    />
    <style>
      .popup:hover { filter: brightness(0.8);}
    </style>
  </svg>
</div>
`;

exports[`Pie labels should plot pie 1`] = `
<div
  style="width: 300px; height: 300px;"
>
  <svg
    class="baseSVG"
    height="100%"
    preserveAspectRatio="xMinYMin meet"
    viewBox="-500 -350 1000 700"
    width="100%"
  >
    <g
      class="canvas"
    >
      <rect
        class="graphArea"
        fill="#fff"
        fill-opacity="1"
        height="700"
        rx="0"
        ry="0"
        stroke="#00c"
        stroke-dasharray="none"
        stroke-opacity="1"
        stroke-width="1"
        width="1000"
        x="-500"
        y="-350"
      />
    </g>
    <g
      class="grids"
    >
      <g
        class="hor-major-Grid"
      />
      <g
        class="hor-minor-Grid"
      />
      <g
        class="ver-major-Grid"
      />
      <g
        class="ver-minor-Grid"
      />
    </g>
    <g
      class="axes"
    >
      <g
        class="horAxis"
      />
      <g
        class="verAxis"
      />
    </g>
    <g
      class="title"
    >
      <text
        dominant-baseline="hanging"
        fill="currentColor"
        font-family="sans-serif"
        font-size="24px"
        font-style="normal"
        font-weight="bold"
        text-anchor="start"
        x="-450"
        y="-315"
      >
        Pie Graph
      </text>
    </g>
    <g
      class="series"
    >
      <g
        class="series0"
        style="color: rgb(27, 158, 119);"
      >
        <g
          class="markers"
          fill="currentColor"
          fill-opacity="0.75"
          stroke="currentColor"
          stroke-dasharray="none"
          stroke-opacity="1"
          stroke-width="1"
        >
          <path
            class="popup"
            d="M-1.2442848271512585e-13,325A325,325,0,0,1,-281.45825622994266,162.49999999999986L0,0Z"
          />
          <path
            class="popup"
            d="M-281.4582562299424,-162.50000000000028A325,325,0,0,1,-111.156546580842,-305.40010175542034L0,0Z"
          />
          <path
            class="popup"
            d="M-281.45825622994266,162.49999999999986A325,325,0,0,1,-281.4582562299424,-162.50000000000028L0,0Z"
          />
          <path
            class="popup"
            d="M-111.156546580842,-305.40010175542034A325,325,0,0,1,5.176144413466479e-13,-325L0,0Z"
          />
          <path
            class="popup"
            d="M1.990051048614449e-14,-325A325,325,0,0,1,320.06251972896763,56.435657741752436L0,0Z"
          />
          <path
            class="popup"
            d="M320.06251972896763,56.435657741752436A325,325,0,0,1,-1.2442848271512585e-13,325L0,0Z"
          />
        </g>
        <g
          class="label"
          fill="#000"
          font-family="sans-serif"
          font-size="16px"
          font-style="normal"
          font-weight="100"
        >
          <text
            class="popup"
            dominant-baseline="middle"
            dx="0.0em"
            dy="0.0em"
            text-anchor="middle"
            transform="translate(-81.25000000000004,140.72912811497127)"
          >
            0.0
          </text>
          <text
            class="popup"
            dominant-baseline="middle"
            dx="0.0em"
            dy="0.0em"
            text-anchor="middle"
            transform="translate(-104.45298657406255,-124.482222006834)"
          >
            1.0
          </text>
          <text
            class="popup"
            dominant-baseline="middle"
            dx="0.0em"
            dy="0.0em"
            text-anchor="middle"
            transform="translate(-162.5,-1.2442848271512585e-13)"
          >
            2.0
          </text>
          <text
            class="popup"
            dominant-baseline="middle"
            dx="0.0em"
            dy="0.0em"
            text-anchor="middle"
            transform="translate(-28.217828870875895,-160.03125986448387)"
          >
            3.0
          </text>
          <text
            class="popup"
            dominant-baseline="middle"
            dx="0.0em"
            dy="0.0em"
            text-anchor="middle"
            transform="translate(124.48222200683395,-104.45298657406263)"
          >
            4.0
          </text>
          <text
            class="popup"
            dominant-baseline="middle"
            dx="0.0em"
            dy="0.0em"
            text-anchor="middle"
            transform="translate(104.45298657406259,124.48222200683396)"
          >
            5.0
          </text>
        </g>
      </g>
      <g
        class="series1"
        style="color: rgb(217, 95, 2);"
      >
        <g
          class="markers"
          fill="currentColor"
          fill-opacity="0.75"
          stroke="currentColor"
          stroke-dasharray="none"
          stroke-opacity="1"
          stroke-width="1"
        >
          <path
            class="popup"
            d="M-1.2442848271512585e-13,325A325,325,0,0,1,-281.45825622994266,162.49999999999986L0,0Z"
          />
          <path
            class="popup"
            d="M-281.4582562299424,-162.50000000000028A325,325,0,0,1,-111.156546580842,-305.40010175542034L0,0Z"
          />
          <path
            class="popup"
            d="M-281.45825622994266,162.49999999999986A325,325,0,0,1,-281.4582562299424,-162.50000000000028L0,0Z"
          />
          <path
            class="popup"
            d="M-111.156546580842,-305.40010175542034A325,325,0,0,1,5.176144413466479e-13,-325L0,0Z"
          />
          <path
            class="popup"
            d="M1.990051048614449e-14,-325A325,325,0,0,1,320.06251972896763,56.435657741752436L0,0Z"
          />
          <path
            class="popup"
            d="M320.06251972896763,56.435657741752436A325,325,0,0,1,-1.2442848271512585e-13,325L0,0Z"
          />
        </g>
        <g
          class="label"
          fill="#000"
          font-family="sans-serif"
          font-size="16px"
          font-style="normal"
          font-weight="100"
        >
          <text
            class="popup"
            dominant-baseline="middle"
            dx="0.0em"
            dy="0.0em"
            text-anchor="middle"
            transform="translate(-81.25000000000004,140.72912811497127)"
          >
            0.0
          </text>
          <text
            class="popup"
            dominant-baseline="middle"
            dx="0.0em"
            dy="0.0em"
            text-anchor="middle"
            transform="translate(-104.45298657406255,-124.482222006834)"
          >
            1.0
          </text>
          <text
            class="popup"
            dominant-baseline="middle"
            dx="0.0em"
            dy="0.0em"
            text-anchor="middle"
            transform="translate(-162.5,-1.2442848271512585e-13)"
          >
            2.0
          </text>
          <text
            class="popup"
            dominant-baseline="middle"
            dx="0.0em"
            dy="0.0em"
            text-anchor="middle"
            transform="translate(-28.217828870875895,-160.03125986448387)"
          >
            3.0
          </text>
          <text
            class="popup"
            dominant-baseline="middle"
            dx="0.0em"
            dy="0.0em"
            text-anchor="middle"
            transform="translate(124.48222200683395,-104.45298657406263)"
          >
            4.0
          </text>
          <text
            class="popup"
            dominant-baseline="middle"
            dx="0.0em"
            dy="0.0em"
            text-anchor="middle"
            transform="translate(104.45298657406259,124.48222200683396)"
          >
            5.0
          </text>
        </g>
      </g>
    </g>
    <g
      class="popup"
    />
    <style>
      .popup:hover { filter: brightness(0.8);}
    </style>
  </svg>
</div>
`;
